File: app/src/main/java/com/riders/thelab/data/remote/dto/weather/WeatherResponse.java
package com.riders.thelab.data.remote.dto.weather;

import com.squareup.moshi.Json;

import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class WeatherResponse {

    @Json(name = "coord")
    Coordinates coordinates;

    @Json(name = "weather")
    List<Weather> weather;

    @Json(name = "base")
    String base;

    @Json(name = "main")
    Main main;

    @Json(name = "visibility")
    int visibility;

    @Json(name = "rain")
    Rain rain;

    @Json(name = "wind")
    Wind wind;

    @Json(name = "clouds")
    Clouds clouds;

    @Json(name = "dt")
    long dt;

    @Json(name = "sys")
    Sys system;

    @Json(name = "timezone")
    int timezone;

    @Json(name = "id")
    int id;

    @Json(name = "name")
    String name;

    @Json(name = "cod")
    int code;
}

File: app/src/main/java/com/riders/thelab/data/remote/dto/weather/WeatherResponse.kt
package com.riders.thelab.data.remote.dto.weather

import kotlinx.serialization.SerialName
import java.io.Serializable

@kotlinx.serialization.Serializable
data class WeatherResponse(

    @SerialName("coord")
    val coordinates: Coordinates,

    @SerialName("weather")
    val weather: List<Weather>,

    @SerialName("base")
    val base: String,

    @SerialName("main")
    val main: Main,

    @SerialName("visibility")
    val visibility: Int = 0,

    @SerialName("rain")
    val rain: Rain,

    @SerialName("wind")
    val wind: Wind,

    @SerialName("clouds")
    val clouds: Clouds,

    @SerialName("dt")
    val dt: Long = 0,

    @SerialName("sys")
    val system: Sys,

    @SerialName("timezone")
    val timezone: Int = 0,

    @SerialName("id")
    val id: Int = 0,

    @SerialName("name")
    val name: String,

    @SerialName("cod")
    val code: Int = 0,
) : Serializable
